dashboard/test/models/sections/section_test.rb

```diff
@@ -71,6 +71,78 @@ class SectionTest < ActiveSupport::TestCase
     end
   end
 
+  test 'changing sharing_disabled to true updates user settings' do
+    student = create :student, sharing_disabled: false
+    section = Section.create @default_attrs
+    assert_creates(Follower) do
+      section.add_student student
+    end
+    section.sharing_disabled = true
+
+    # Student update happens on before save
+    section.save
+    student.reload
+    assert student.sharing_disabled?
+  end
+
+  test 'changing sharing_disabled to false maintains user settings' do
+    student = create :student, sharing_disabled: false
+    section = Section.create @default_attrs
+    assert_creates(Follower) do
+      section.add_student student
+    end
+
+    # Student update happens on before save
+    section.save
+    student.reload
+    assert !student.sharing_disabled?
+  end
```

Comment:
sharing_disabled wasn't changed here. The section is saved with no changes. Maybe we should initialize this section with sharing_disabled true:

```ruby
section = create :section, sharing_disabled: true
```